### PR TITLE
[luci] Introduce shape_status to CircleNode

### PR DESCRIPTION
**Pull request by seanshpark**
This will introduce shape_status to CircleNode as node shape status

ONE-DCO-1.0-ONE-DCO-1.0-Signed-off-by: SaeHie Park <saehie.park@gmail.com>

### CODE REVIEW

**Comment by seanshpark** [comment]
For #1610, from draft #1363

**Comment by seanshpark** [comment]
@mhs4670go @llFreetimell @jinevening PTAL

File: compiler/luci/lang/include/luci/IR/PropertyShapeStatus.h [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+#ifndef __LUCI_IR_PROPERTY_SHAPE_STATUS_H__
+#define __LUCI_IR_PROPERTY_SHAPE_STATUS_H__
+
+namespace luci
+{
+
+/**
+ * @brief  ShapeStatus is to remember circle node shape status.
+ * @note   This is not an attribute from the file but inner status of a node.
+ *         Shape with [] is scalar but somtimes it acts as dynamic shape.
```
Comment (llFreetimell) [inline compiler/luci/lang/include/luci/IR/PropertyShapeStatus.h:26]: typo : sometimes

File: compiler/luci/lang/include/luci/IR/PropertyShapeStatus.h [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ */
+enum class ShapeStatus
+{
+  UNDEFINED, // Shape status is undefined
+
+  NOSHAPE, // shape is unknown; to distinguish from scala
```
Comment (llFreetimell) [inline compiler/luci/lang/include/luci/IR/PropertyShapeStatus.h:32]: typo : scalar

**Pull request merged**